emu/src/cpu/arm7tdmi.rs

```diff
@@ -224,6 +238,43 @@ impl Arm7tdmi {
         }
     }
 
+    fn load_store_register_offset(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let load_store: LoadStoreKind = op_code.get_bit(11).into();
+        let byte_word: ReadWriteKind = op_code.get_bit(10).into();
+        let ro = op_code.get_bits(6..=8);
+        let ro = self.registers.register_at(ro.try_into().unwrap());
+
+        let rb = op_code.get_bits(3..=5);
+        let rb = self.registers.register_at(rb.try_into().unwrap());
+
+        let rd: usize = op_code.get_bits(0..=2).try_into().unwrap();
+
+        let address: usize = rb.wrapping_add(ro).try_into().unwrap();
+
+        let mut mem = self.memory.lock().unwrap();
+
+        match (load_store, byte_word) {
+            (LoadStoreKind::Store, ReadWriteKind::Byte) => {
+                let rd = (self.registers.register_at(rd) & 0xFF) as u8;
```

Comment:
`self.registers.register_at(rd)` returns a u32, why `& 0xFF` and cast as u8? You are not considering 8 bits here